wellen/src/ghw/hierarchy.rs

```diff
@@ -1545,10 +1543,22 @@ impl IntRange {
         opt.unwrap_or(Self(RangeDir::To, i32::MIN as i64, i32::MAX as i64))
     }
 
+    fn start(&self) -> i64 {
+        match self.0 {
+            RangeDir::To => self.1,
+            RangeDir::Downto => self.2,
+        }
+    }
+
+    fn end(&self) -> i64 {
+        match self.0 {
+            RangeDir::To => self.2,
+            RangeDir::Downto => self.1,
+        }
+    }
```

Comment:
Could you add some unittests inside this file to test this new functionality?

So, add something like this to the end of `hierarchy.rs`:

```rs
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_int_range() {
       // TODO: create some ranges and test the `start`, `end` and `range` functionality
    }
```

Reply:
Sure! (I pushed a new version with first comment only before seeing this...)

Not sure how to handle empty ranges though (1 to 0, 0 downto 1). Nor if it matters in practice. I guess `start` and `end` doesn't really make sense then. But `len` should probably clamp to 0?